presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ReplicateSemiJoinInDelete.java

```diff
@@ -66,7 +62,13 @@ public PlanNode visitSemiJoin(SemiJoinNode node, RewriteContext<Void> context)
                     node.getSemiJoinOutput(),
                     node.getSourceHashSymbol(),
                     node.getFilteringSourceHashSymbol(),
-                    Optional.of(targetJoinDistributionType));
+                    Optional.empty());
```

Comment:
if `! isDeleteQuery`, the optimizer should not change anything; here it empties distribution type